frontend/src/components/RideStatus/SModal.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useRef, useEffect } from 'react';
+import moment from 'moment';
+import cn from 'classnames';
+import { Ride } from '../../types';
+import styles from './smodal.module.css';
+import ProgressBar from './ProgressBar';
+
+type SModalProps = {
+  isOpen: boolean;
+  close: () => void;
+  ride: Ride | undefined
+};
+
+const SModal = ({
+  isOpen,
+  close,
+  ride,
+}: SModalProps) => {
+  function useOutsideAlerter(ref: any) {
+    useEffect(() => {
+      function handleClickOutside(event: any) {
+        if (ref.current && !ref.current.contains(event.target)) {
+          close();
+        }
+      }
+
+      document.addEventListener('mousedown', handleClickOutside);
+      return () => {
+        document.removeEventListener('mousedown', handleClickOutside);
+      };
+    }, [ref]);
+  }
+  const wrapperRef = useRef(null);
+  useOutsideAlerter(wrapperRef);
+
+  return (
+    <div>
+      {isOpen && ride && (
+        <div className={styles.modal} ref={wrapperRef}>
+          <div className={styles.body}>
+            <button className={styles.close} onClick={close}>✕</button>
+            <h3 className={styles.title}>
+              {`${ride.rider.firstName} ${ride.rider.lastName}`}
+            </h3>
+            <p>Status Updates</p>
+            <div className={styles.bar}>
+              <ProgressBar status={ride.status} late={ride.late} />
+            </div>
+            <div className={styles.row}>
+              <div className={styles.column}>
+                <p>{moment(new Date(ride.startTime)).format('h:mm a')}</p>
+                <div className={styles.location}>
+                  <span className={cn(styles.tag, styles[ride.startLocation.tag || ''])}>{ride.startLocation.tag}</span>
+                  <p>{ride.startLocation.name}</p>
+                </div></div>
```

Comment:
can you make this more responsive for longer addresses? also add the gray bg like in the design
![image](https://user-images.githubusercontent.com/20343975/111095031-ef8f8d00-8512-11eb-9f5f-217874059e04.png)

Reply:
this is 2/27 btw

Reply:
Nice catches. 👍